internal/providers/github/common.go

```diff
@@ -64,6 +64,9 @@ var (
 	ErrBranchNotFound = errors.New("branch not found")
 	// ErrNoPackageListingClient denotes if there is no package listing client available
 	ErrNoPackageListingClient = errors.New("no package listing client available")
+	// ErroNoCheckPerissions is a fixed error returned when the credentialed
+	// identity has not been authorized to use the checks API
+	ErroNoCheckPerissions = errors.New("missing permissions: check")
```

Comment:
Do we have the permission by default with the GH App? I think we don't with the OAuth token we request..

Reply:
I'm adding these APIs so that we can get rid of the comment spam.

Currently, we don't have permissions for the checks, this is why I created a custom error that traps it, so that we can handle runs without them. I think we will have to support running without it, at least for a while.

Reply:
I believe that the checks API is _only_ available to GitHub Apps:

https://docs.github.com/en/rest/guides/using-the-rest-api-to-interact-with-checks?apiVersion=2022-11-28

> If you aren't building a GitHub App, you might be interested in using the REST API to interact with [commit statuses](https://docs.github.com/en/rest/commits#commit-statuses).

Reply:
That's not to say that we should limit ourselves to the lowest common denominator.

Reply:
Yes, this is why the OSV evaluator uses commit statuses (the OSV evaluator predates the GH app)